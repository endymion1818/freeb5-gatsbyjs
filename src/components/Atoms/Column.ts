import styled from 'styled-components'

export interface IColumnProps {
  /**
   * alignment along vertical axis
   * @default center
   */
  verticalAlign?: string
  /**
   * gap above top of item
   * @default 0
   */
  bufferTop?: string
  /**
   * gap below bottom of item
   * @default 0
   */
  bufferBottom?: string
  /**
   * text alignment
   * @default left
   */
  textAlign?: string
}

const Column = styled.div<IColumnProps>`
    display: flex;
    flex-direction: column;

    align-self: ${({ verticalAlign }) => verticalAlign};

    ${({ bufferTop }) => bufferTop && `margin-top: ${bufferTop};`}
    ${({ bufferBottom }) => bufferBottom && `margin-bottom: ${bufferBottom};`}

    ${({ textAlign }) => textAlign && `text-align: ${textAlign};`}

    > h1,
    > h2,
    > h3,
    > h4,
    > h5,
    > h6 {
        flex: 1;
    }
    > p {
        flex: 2;

        & ~ div[class*='Button'] {
        flex: 0;
        }
    }
    > img {
        width: 100%;
    }
`

Column.defaultProps = {
  verticalAlign: 'center',
  textAlign: 'left',
  bufferTop: '0rem',
  bufferBottom: '0rem',
}
export default Column
