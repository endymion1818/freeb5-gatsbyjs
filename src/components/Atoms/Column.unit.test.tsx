import { cleanup, render } from "@testing-library/react";
import React from "react";
import Column from "./Column";

const ColumnDefaultProps = {
  children: "test"
};

describe("<Column />", () => {
  beforeEach(cleanup);
  it("Should render", () => {
    const { container } = render(<Column {...ColumnDefaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
