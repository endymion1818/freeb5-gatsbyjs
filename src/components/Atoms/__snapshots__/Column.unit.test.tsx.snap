// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Column /> Should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin-top: 0rem;
  margin-bottom: 0rem;
  text-align: left;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 > p {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c0 > p ~ div[class*='Button'] {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
}

.c0 > img {
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    test
  </div>
</div>
`;
